python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -176,35 +176,101 @@ def test_basic(ray_start_regular):
         del result
 
 
-def test_basic_destruction(ray_start_regular):
-    a = Actor.remote(0)
-    with InputNode() as i:
-        dag = a.echo.bind(i)
+class TestDAGRefDestruction:
+    def test_basic_destruction(self, ray_start_regular):
+        a = Actor.remote(0)
+        with InputNode() as i:
+            dag = a.echo.bind(i)
+        compiled_dag = dag.experimental_compile()
 
-    compiled_dag = dag.experimental_compile()
+        try:
+            for i in range(3):
+                val = np.ones(100) * i
+                ref = compiled_dag.execute(val)
+                # Since ref.get() is not called, the destructor releases its native
+                # buffer without deserializing the value. If the destructor fails to
+                # release the buffer, the subsequent DAG execution will fail due to
+                # memory leak.
+                del ref
+        except RayChannelTimeoutError:
+            pytest.fail(
+                "The native buffer associated with the CompiledDAGRef was not "
+                "released upon destruction."
+            )
 
-    try:
-        for i in range(3):
-            val = np.ones(100) * i
-            ref = compiled_dag.execute(val)
-            # Since ref.get() is not called, the destructor releases its native
-            # buffer without deserializing the value. If the destructor fails to
-            # release the buffer, the subsequent DAG execution will fail due to
-            # memory leak.
-            del ref
-    except RayChannelTimeoutError:
-        pytest.fail(
-            "The native buffer associated with the CompiledDAGRef was not "
-            "released upon destruction."
-        )
+        # Ensure that subsequent DAG executions do not fail due to memory leak
+        # and the results can be retrieved by ray.get().
+        val = np.ones(100)
+        ref = compiled_dag.execute(val)
+        result = ray.get(ref)
+        assert (result == val).all()
+        del ref
 
-    # Ensure that subsequent DAG executions do not fail due to memory leak
-    # and the results can be retrieved by ray.get().
-    val = np.ones(100)
-    ref = compiled_dag.execute(val)
-    result = ray.get(ref)
-    assert (result == val).all()
-    del ref
+    def test_get_ref_before_destructed_ref(self, ray_start_regular):
+        a = Actor.remote(0)
+        with InputNode() as inp:
+            dag = a.inc.bind(inp)
+        compiled_dag = dag.experimental_compile(_max_inflight_executions=3)
+        ref = compiled_dag.execute(1)
+        ref2 = compiled_dag.execute(1)
+        del ref2
+        # Test that ray.get() on ref still works properly even if
+        # ref2 (corresponding to a later execution) is destructed first
+        time.sleep(0.1)
```

Comment:
Is there a better way to ensure that the destructor is called? By the way, if sleep is needed to ensure `__del__` is called, why don’t the following tests call sleep?